plugins/modules/iosxr_bgp_address_family.py

```diff
@@ -0,0 +1,614 @@
+                backdoor_route_policy:
+                  type: str
+                  description: Specify a BGP backdoor route.
+                route_policy:
+                  type: str
+                  description: Route-policy to modify the attributes.
+            nexthop: &nexthop
+              type: dict
+              description: Nexthop
+              suboptions:
+                resolution_prefix_length_minimum:
+                  type: int
+                  description: Set minimum prefix-length for nexthop resolution.
+                  choices: [0,32]
+                route_policy:
+                  type: str
+                  description: Policy to filter out nexthop notification.
+                trigger_delay_critical:
+                  description: For critical notification
+                  type: int
+                trigger_delay_non_critical:
+                  type: int
+                  description: For non critical notification.
+            optimal_route_reflection: &optimal_rr
+              type: dict
+              description: Configure optimal-route-reflection group.
+              suboptions:
+                group_name:
+                  type: str
+                  description: ORR group name - maximum 32 characters.
+                primary_address:
+                  type: str
+                  description: IPv4 primary address.
+                secondary_address:
+                  type: str
+                  description: IPv4 secondary address
+            permanent_network_route_policy:
+              type: str
+              description: Name of the policy.
+            retain_local_label:
+              type: int
+              description: Label retention time in minutes <3-60>.
+            table_policy: &table_policy
+              type: str
+              description: Configure policy for installation of routes to RIB.
+            update:
+              type: dict
+              description: BGP Update generation configuration.
+              suboptions:
+                limit:
+                  type: dict
+                  description: Update limit.
+                  suboptions:
+                    sub_group:
+                      type: dict
+                      description: Update limit for address-family.
+                      suboptions:
+                        ibgp:
+                          type: int
+                          description: Update limit for iBGP sub-groups<1-512.
+                        ebgp:
+                          type: int
+                          description: Update limit for eBGP sub-groups<1-512.
+                    address_family:
+                      type: int
+                      description: Update limit for sub-groups.
+                wait_install:
+                  type: bool
+                  description: Wait for route install.
+            redistribute: &redistribute
```

Comment:
This should also be a list.

Reply:
Please refer to how redistribute is handled from either bgp or ospf modules.

Reply:
ack,made changes